pkg/auth/dependency/authenticator/password/provider.go

```diff
@@ -41,14 +41,22 @@ func (p *Provider) List(userID string) ([]*Authenticator, error) {
 }
 
 func (p *Provider) New(userID string, password string) (*Authenticator, error) {
-	err := p.isPasswordAllowed(userID, password)
-	if err != nil {
-		return nil, err
-	}
+	var hash []byte
+	// NOTE: if there is no existing password authenticator when user call
+	// add login id api, api will create authenticator with empty password
+	// we use a empty password hash to make it unable to be used to login.
+	if password != "" {
```

Comment:
This logic cannot distinguish between empty password and no password. I suppose we will not support allowing user to login/signup/updated with a empty password.

Reply:
Yes, I think we don't support empty password. So the empty string here means no password actually. I updated the comment 2c64d83, see if it looks clear:)